src/group_theory/perm/sign.lean

```diff
@@ -617,6 +616,16 @@ let ⟨a, ha⟩ := hg.2 x hx in
 let ⟨b, hb⟩ := hg.2 y hy in
 ⟨b - a, by rw [← ha, ← mul_apply, ← gpow_add, sub_add_cancel, hb]⟩
 
+lemma exists_pow_eq_of_is_cycle [decidable_eq β] [fintype β] {f : perm β}
```

Comment:
Shouldn't this be in `perm/cycles` instead?

Reply:
Ah, I guess the lemma above is also in the wrong place

Reply:
is_cycle is defined in this file, which I think is why the lemma above is in this file

Reply:
It would probably make sense to move the definition and the lemmas to the other file - are you ok to do that in this PR, or do you want to punt it for someone else to do some other time?

Reply:
There seems to be a slight problem: perm/cycles imports perm/sign, and perm/sign uses these results. So you can't just move this stuff to the other file. You would have to reverse the import order.

Reply:
I couldn't find that - where does `perm/sign` use these results?

Reply:
To be clear, my suggestion would be to move all of the following to the other file:
```
is_cycle
is_cycle_swap
is_cycle_inv
exists_gpow_eq_of_is_cycle
exists_pow_eq_of_is_cycle
is_cycle_swap_mul_aux₁ 
is_cycle_swap_mul_aux₂
eq_swap_of_is_cycle_of_apply_apply_eq_self
is_cycle_swap_mul
sign_cycle
```

Reply:
For instance, the `sign_cycle` lemma uses `is_cycle`, so if you moved the definition of `is_cycle` to `perm/cycles` then things would break

Reply:
Ah, I see what you're saying now

Reply:
Ok, I made an attempt at moving the lemmas over